src/ORM/DataQuery.php

```diff
@@ -1038,13 +1069,29 @@ protected function joinManyManyRelationship(
     ) {
         $schema = DataObject::getSchema();
 
+        /** @var Versioned $versioned */
+        $versioned = Injector::inst()->get(Versioned::class);
```

Comment:
Sorry @mfendeksilverstripe - this bit probably needs the same guards you did for the other bit of code.

Reply:
Good spotting, thanks :)